src/phases/attempt-capture-phase.ts

```diff
@@ -114,13 +121,18 @@ export class AttemptCapturePhase extends PokemonPhase {
                   if (!pokemon.species.isObtainable()) {
                     shakeCounter.stop();
                     this.failCatch(shakeCount);
-                  } else if (shakeCount++ < 3) {
-                    if (pokeballMultiplier === -1 || pokemon.randSeedInt(65536) < y) {
+                  } else if (shakeCount++ < (isCritical ? 1 : 3)) {
+                    // Shake check (skip check for critical or guaranteed captures, but still play the sound)
+                    if (pokeballMultiplier === -1 || isCritical || modifiedCatchRate >= 255 || pokemon.randSeedInt(65536) < shakeProbability) {
                       this.scene.playSound("se/pb_move");
                     } else {
                       shakeCounter.stop();
                       this.failCatch(shakeCount);
                     }
+                  } else if (isCritical && pokemon.randSeedInt(65536) >= shakeProbability) {
+                    // Above, perform the one shake check for critical captures after the ball shakes once
```

Comment:
```suggestion
                  } else if (pokeballMultiplier !== -1 && pokemon.randSeedInt(65536) >= shakeProbability) {
                    // Perform shake check after final ball shake (for both critical and non-critical captures)
```
This will be the 4th shake check for non-critical captures

Reply:
Just realized, yes, we don't do a 4th shake check. I don't think we should though, catching is hard enough as is (this PR was meant to alleviate that) and I imagine it would generate bug reports if mons started breaking out after 3 shakes.